core/src/core/transaction.rs

```diff
@@ -971,24 +982,40 @@ impl Transaction {
 /// and outputs.
 pub fn cut_through(inputs: &mut Vec<Input>, outputs: &mut Vec<Output>) -> Result<(), Error> {
 	// assemble output commitments set, checking they're all unique
-	let mut out_set = HashSet::new();
-	let all_uniq = { outputs.iter().all(|o| out_set.insert(o.commitment())) };
-	if !all_uniq {
+	outputs.sort_unstable_by(|v1, v2| v1.commitment().0.cmp(&v2.commitment().0));
```

Comment:
Can we leverage `sort_unstable_by_key` here and get rid of the need to `cmp` nested stuff on v1 and v2 explicitly here? Might read better.

Reply:
We should be able to sort on `v1.commitment()` and not need to go deeper than that with `v1.commitment().0` ?

Reply:
I think that there's no Ord impl for commitment?

sort_unstable_by_key doesn't work well because I think we're technically comparing the &[u8] slices, not the [u8; 32]'s so we want to return a ref. Not sure... maybe it does work.

Reply:
:+1:

Reply:
See below - there's no `Ord` for commitment, but there _is_ an `Ord` for `Hashable` things (including Inputs/Outputs/TxKernels).